### PR TITLE
[ESI-14406] Added converter for code generation for golang extensions.

### PR DESCRIPTION
**Pull request by jkmar**
Converter generates code from yaml schemas.
Generated code:

- Definition of structs representing objects from each schema
- Interfaces for getters and setters for these objects
- Implementation of these interfaces by pointers to generated structs
- Interfaces that can be extended
- Constructors for objects with default values
- Database functions for generated structs (fetch, list)

### CODE REVIEW

**Comment by morrisson** [comment]
Add docs, please :)

- overview
- how to use
- example (yaml to go)
- limitation (if applicable)

**Comment by morrisson** [comment]
@jkmar Any updates about docs?

**Comment by jkmar** [comment]
@morrisson I've added docs.

**Comment by p-kozlowski** [comment]
looks good, but please rebase

**Comment by jkmar** [comment]
@morrisson @marcin-ptaszynski please review.

File: cli/cli.go [hunk 1]
```diff
@@ -458,6 +461,50 @@ func getCreateInitialMigrationSubcommand() cli.Command {
...
+	* Constructors for objects with default values
+	* Database functions for generated structs (fetch, list)
+ARGUMENTS:
+	There is one argument - path to file with yaml schemas
+`,
+		Flags: []cli.Flag{
+			cli.StringFlag{Name: "goext-package", Value: "goext", Usage: "Package name for golang extension interfaces"},
+			cli.StringFlag{Name: "crud-package", Value: "goodies", Usage: "Package name for crud functions"},
+			cli.StringFlag{Name: "raw-package", Value: "resources", Usage: "Package name for raw structs"},
+			cli.StringFlag{Name: "interface-package", Value: "esi", Usage: "Package name for interfaces"},
```
Comment (morrisson) [inline cli/cli.go]: Do not expose `esi` to OSS gohan.

File: docs/converter.md [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+	* Database functions for generated structs (fetch, list)
+ARGUMENTS:
+	There is one argument - path to file with yaml schemas
+
+
+OPTIONS:
+   --goext-package "goext"	Package name for golang extension interfaces
+   --crud-package "goodies"	Package name for crud functions
+   --raw-package "resources"	Package name for raw structs
+   --interface-package "esi"	Package name for interfaces
```
Comment (morrisson) [inline docs/converter.md]: ditto

File: docs/converter.md [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+    }   
+}
+```
+
+- [output]_genereted_interface.go<br>
+
+File with interfaces of getters and setters for structs that are in [output]_raw.go file
+
+```go
+package esi
```
Comment (morrisson) [inline docs/converter.md]: ditto

File: cli/cli.go (hunk 1 above)
Comment (jkmar) [reply cli/cli.go]: @morrisson I've changed default interface package name to "interfaces".

**Comment by przemyslaw-dobrowolski-cl** [comment]
@morrisson A kind remainder about this PR - could you review it again?

**Comment by przemyslaw-dobrowolski-cl** [comment]
@morrisson could you review this PR?

File: cli/cli.go [hunk 4]
```diff
@@ -458,6 +461,50 @@ func getCreateInitialMigrationSubcommand() cli.Command {
 	}
 }
 
+func getGenerateGoextCommand() cli.Command {
```
Comment (przemyslaw-dobrowolski-cl) [inline cli/cli.go]: maybe this should be renamed to getConverter command

Comment (jkmar) [reply cli/cli.go]: this name is already used by different command

Comment (jkmar) [reply cli/cli.go]: I've change the name to getConverterCommand

**Pull request merged**